### PR TITLE
Repackaging Azure Active Directory with Detections migration

### PR DESCRIPTION
**Pull request by v-vdixit**
Required items, please complete
   
   Change(s):
   - Updating queries of analytic rules to be migrated to AAD  solution

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationInboundDirectSettingsChanged.yaml [hunk 1]
```diff
@@ -29,8 +29,16 @@ query: |
   | where B2BType = TargetResources[0].modifiedProperties[0].displayName has "b2bDirectInbound"
```
Comment (devikamehra) [inline Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationInboundDirectSettingsChanged.yaml:29]: There is only b2bDirectConnectInbound

File: Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationOutboundDirectSettingsChanged.yaml [hunk 2]
```diff
@@ -29,8 +29,16 @@ query: |
   | where B2BType = TargetResources[0].modifiedProperties[0].displayName has "b2bDirectOutbound"
```
Comment (devikamehra) [inline Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationOutboundDirectSettingsChanged.yaml:29]: Again, only b2bDirectConnectOutbound

File: Detections/AuditLogs/GuestAccountsAddedinAADGroupsOtherThanTheOnesSpecified.yaml [hunk 3]
```diff
@@ -29,16 +29,31 @@ query: |
...
-  | mv-expand AADGroupId = TargetResources[1].id to typeof(string)
+  | mv-apply TargetResource = TargetResources on 
+    (
+        where TargetResource.type =~ "ObjectId"
+        | extend AADGroupId = trim('"',tostring(TargetResource.id))
+    )
   | where AADGroupId !in (GroupIDs)
+  | extend Name = tostring(split(InvitedUser,'@',0)[0]), UPNSuffix = tostring(split(InvitedUser,'@',1)[0])
 entityMappings:
   - entityType: Account
```
Comment (devikamehra) [inline Detections/AuditLogs/GuestAccountsAddedinAADGroupsOtherThanTheOnesSpecified.yaml:51]: InitiatedByActionUserInformation can also be mapped as one entity of type account.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Please verify queries for different suggestions. Also, please check if any of the query is duplicate with AAD solution content. We should not add those in the solution.

File: Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationAdded.yaml [hunk 4]
```diff
@@ -27,21 +27,32 @@ query: |
...
+        where Property.displayName =~ "tenantId"
+        | extend ExtTenantIDAdded = trim('"',tostring(Property.newValue))
+    )
   | where ExtTenantIDAdded !in (ExpectedTenantIDs)
+  | extend Name = tostring(split(InitiatedByActionUserInformation,'@',0)[0]), UPNSuffix = tostring(split(InitiatedByActionUserInformation,'@',1)[0])
 entityMappings:
   - entityType: Account
     fieldMappings:
-      - identifier: FullName
+      - identifier: Name
```
Comment (devikamehra) [inline Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationAdded.yaml:45]: This is not a valid Account value. Use custom details for storing this value.

File: Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationInboundCollaborationSettingsChanged.yaml [hunk 5]
```diff
@@ -26,20 +26,32 @@ query: |
   // When Access Type in modified inbound settings value is 1 that means that now access is allowed. When Access Type in modified inbound settings value is 2 that means that now access is blocked. 
   AuditLogs
   | where OperationName has "Update a partner cross-tenant access setting"
-  | where B2BType = TargetResources[0].modifiedProperties[0].displayName has "b2bCollaborationInbound"
+  | mv-apply TargetResource = TargetResources on
+    (
+        where TargetResource.type =~ "Policy"
+        | extend Properties = TargetResource.modifiedProperties
+    )
+  | where B2BType = Properties.displayName has "b2bCollaborationInbound"
```
Comment (devikamehra) [inline Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationInboundCollaborationSettingsChanged.yaml:34]: This is not required. Move lines 37 to 42 here.

File: Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationInboundCollaborationSettingsChanged.yaml [hunk 6]
```diff
@@ -26,20 +26,32 @@ query: |
...
+        | extend PremodifiedInboundSettings = trim('"',tostring(Property.oldValue)),
+                 ModifiedInboundSettings = trim(@'"',tostring(Property.newValue))
+    )
   | where PremodifiedInboundSettings != ModifiedInboundSettings
+  | extend Name = tostring(split(InitiatedByActionUserInformation,'@',0)[0]), UPNSuffix = tostring(split(InitiatedByActionUserInformation,'@',1)[0])
 entityMappings:
   - entityType: Account
     fieldMappings:
       - identifier: Name
         columnName: InitiatedByActionUserInformation
```
Comment (devikamehra) [inline Detections/AuditLogs/Cross-tenantAccessSettingsOrganizationInboundCollaborationSettingsChanged.yaml:49]: This should be Name

File: Detections/AuditLogs/GuestAccountsAddedinAADGroupsOtherThanTheOnesSpecified.yaml [hunk 7]
```diff
@@ -29,23 +29,41 @@ query: |
...
     fieldMappings:
-      - identifier: FullName
+      - identifier: Name
         columnName: InvitedUser
+      - identifier: UPNSuffix
+        columnName: UPNSuffix
   - entityType: Account
     fieldMappings:
       - identifier: Name
         columnName: InitiatedByActionUserInformation
```
Comment (devikamehra) [inline Detections/AuditLogs/GuestAccountsAddedinAADGroupsOtherThanTheOnesSpecified.yaml:61]: Map "Name" here.

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Review by devikamehra — DISMISSED** [review]

**Review by v-rbajaj — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**